Sources/ClientGtk/ServerListView.swift

```diff
@@ -1,15 +1,20 @@
 import SwiftCrossUI
 import DeltaCore
 
-enum DetailState {
-  case addition
-  case server(ServerDescriptor)
-  case error(String)
+indirect enum DetailState {
+  case server(_ index: Int)
+  case adding
+  case editing(_ index: Int)
+
+  case error(_ message: String, returnState: DetailState?)
```

Comment:
You can probably just change `returnState` to a plain `DetailState` since it's never set to `nil` as far as I can see. It's best not to try and assume what will be needed in the future when designing APIs (for small APIs such as this state enum).